library/src/main/java/com/novoda/downloadmanager/LiteDownloadBatchStatus.java

```diff
@@ -116,13 +113,8 @@ private void updateStatus(Status status, DownloadsBatchStatusPersistence persist
         persistence.updateStatusAsync(downloadBatchId, status);
     }
 
-    @Nullable
     @Override
     public DownloadError.Error getDownloadErrorType() {
-        if (downloadError != null) {
-            return downloadError.error();
-        } else {
-            return null;
-        }
+        return downloadError.or(new DownloadError(DownloadError.Error.UNKNOWN)).error();
```

Comment:
I think this needs to still be `@Nullable` but the errors are not stored by the persistence layer so when we come back to the app and an error is displayed it will appear as `UNKNOWN` or `null` 😢